Foxtrot/Foxtrot/Checker/VisibilityHelper.cs

```diff
@@ -37,6 +41,22 @@ private void ReInitialize(Method asThisMember)
             this.AsThisMember = asThisMember;
         }
 
+        private static bool IsAttribute(TypeNode declaringType)
+        {
+            if (declaringType == null || declaringType.BaseType == null)
+            {
+                return false;
+            }
+
+            if (declaringType == systemAttributeType.Value || declaringType.BaseType == systemAttributeType.Value)
+            {
+                return true;
+            }
+            
+
+            return IsAttribute(declaringType.BaseType);
+        }
+
```

Comment:
You are right, thanks. Otherwise it might be less efficient than recursive version...